### PR TITLE
[24.2] Final changes for 24.2rc1

### PR DESCRIPTION
**Pull request by fanquake**
Final changes to tag a `24.2rc1`.
Bumps version numbers, man pages, adds release notes etc.

### CODE REVIEW

**Review by stickies-v — APPROVED** [review]
ACK 43f853d54918ff91268778006cc2d75aa26bd33a modulo two small fixes

I verified the release notes correspond with the the commits since 3116ccd790e76de8f64d2ef9aa5a2641c15bbd8b, and the manpages and bitcoin.conf correspond with my output.

File: doc/release-notes.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
-24.1 Release Notes
+24.2rc1 Release Notes
 ==================
 
-Bitcoin Core version 24.1 is now available from:
+Bitcoin Core version 24.2 is now available from:
```
[the remark this answers is not in the document]
Comment (fanquake) [reply doc/release-notes.md]: Fixed.

File: doc/release-notes.md [hunk 2]
```diff
@@ -37,63 +37,40 @@ Core should also work on most other Unix-like systems but is not as
...
-- #26515 rpc: Require NodeStateStats object in getpeerinfo
-- #27279 doc: fix/improve warning helps in {create,load,unload,restore}wallet
-- #27468 rest: avoid segfault for invalid URI
+- #27727 rpc: Fix invalid bech32 address handling
 
 ### Build System
 
-- #26944 depends: fix systemtap download URL
-- #27462 depends: fix compiling bdb with clang-16 on aarch64
+- #28410 depends: xcb-proto 1.15.2
```
[the remark this answers is not in the document]
Comment (fanquake) [reply doc/release-notes.md]: Fixed.

**Review by stickies-v — APPROVED** [review]
ACK d8340dd03bfaee1a888478803ce4f0cfeb0cc98d

**Comment by achow101** [comment]
ACK 43f853d54918ff91268778006cc2d75aa26bd33a

**Pull request merged**